### PR TITLE
Studio revamp

### PR DESCRIPTION
**Pull request by proksh**
- Added subfolders in the studio app
> components - contains all the components related to app
> pages - contains main pages only
> api - contains api stuff
> const - contains all the types, constants, and functions that can be used in the app
> storybook - contains storybook components
> style - all the style files

- Revamped the studio design
- Added search components panel
- Added design system
- Changed the structure of the Action modal
- Changed drag-and-drop UI and package for command reorder

### CODE REVIEW

**Comment by Leland-Takamine** [comment]
Nice work Proksh!

Noticed that the code gets cut off below the action icons on the right:

image

File: maestro-studio/web/src/pages/InteractPage.tsx [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import { useEffect, useState } from "react";
+import clsx from "clsx";
+
+import InteractableDevice from "../components/device-and-device-elements/InteractableDevice";
+import ReplView from "../components/commands/ReplView";
+import { DeviceScreen, UIElement } from "../const/models";
+import { API, wait } from "../api/api";
+// import { ActionModal } from "../components/depreciated(toremove)/ActionModal";
```
Comment (Leland-Takamine) [inline maestro-studio/web/src/pages/InteractPage.tsx:8]: Comment ^

File: maestro-studio/web/src/pages/InteractPage.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+  return (
+    <div className="flex h-full overflow-hidden">
+      {showElementsPanel && (
+        <ElementsPanel
+          hoveredElement={hoveredElement}
+          setHoveredElement={setHoveredElement}
+          deviceScreen={deviceScreen}
+          onElementSelected={(element: UIElement | null) => {
+            console.log(element);
```
Comment (Leland-Takamine) [inline maestro-studio/web/src/pages/InteractPage.tsx:53]: Remove log

File: maestro-studio/web/src/components/commands/CommandRow.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+export default function CommandRow({
+  command,
+  selected,
+  onClick,
+  isDragging,
+}: CommandRowProps) {
+  // const [value, setValue] = useState(command.yaml);
+
+  console.log(command.yaml);
```
Comment (Leland-Takamine) [inline maestro-studio/web/src/components/commands/CommandRow.tsx:26]: Remove log

File: maestro-studio/web/src/components/commands/CommandRow.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+  isDragging: boolean;
+}
+
+export default function CommandRow({
+  command,
+  selected,
+  onClick,
+  isDragging,
+}: CommandRowProps) {
+  // const [value, setValue] = useState(command.yaml);
```
Comment (Leland-Takamine) [inline maestro-studio/web/src/components/commands/CommandRow.tsx:24]: Remove comment

File: maestro-studio/web/src/components/commands/CommandRow.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+          size="sm"
+          checked={selected}
+          className="pointer-events-none"
+        />
+      </div>
+      <div className="flex-grow">
+        <pre className="font-mono cursor-default overflow-x-scroll text-sm text-gray-900 dark:text-white py-0.5 w-full">
+          {command.yaml}
+        </pre>
+        {/* <CommandInput
```
Comment (Leland-Takamine) [inline maestro-studio/web/src/components/commands/CommandRow.tsx:47]: commented code

**Comment by proksh** [comment]
@Angelk90 

Updated
- Fixed the browser-related issues you mentioned
- About screen resizing, the app isn't responsive right now, but I did sort the resizing issues till larger tablet devices.

Left out
- Edit an already added field that doesn't have an API right now. It will be updated later
- About the margin on the button, I don't think it's necessary because the button doesn't have a background in the resting state. Let me know if there is any particular reason for mentioning it that I am missing.

**Comment by proksh** [comment]
Hey @Angelk90 

I agree with both of your points.

- Regarding the first point, I believe it would be more appropriate to address the issues related to size and responsiveness in a separate pull request at a later stage.

- As for the second point, I share the same sentiment about dark-mode colors. :) Currently, the app is using the default theme provided by Tailwind. Given that implementing changes for dark mode would require significant modifications, it would be more feasible to handle this in a different pull request as well.

**Review by Leland-Takamine — APPROVED** [review]

**Comment by Leland-Takamine** [comment]
Text color issue:
image

**Review by axelniklasson — APPROVED** [review]
Noticed a bug with the export modal in light mode -- aside from that it looks great to me, so approving for now assuming this will be fixed 🙌 

image

**Comment by proksh** [comment]
@Angelk90 Nice catch.

Pushed the fix for this.

**Comment by proksh** [comment]
@Angelk90 It should be fixed now.

**Pull request closed without merging**